node/src/accountant/db_access_objects/sent_payable_dao.rs

```diff
@@ -0,0 +1,844 @@
+        subject
+            .insert_new_records(&vec![tx1.clone(), tx2.clone(), tx3.clone()])
+            .unwrap();
+        let hashset = HashSet::from([tx1.hash, tx2.hash]);
+
+        let result = subject.delete_records(&hashset);
+
+        let remaining_records = subject.retrieve_txs(None);
+        assert_eq!(result, Ok(()));
+        assert_eq!(remaining_records, vec![tx3]);
+    }
+
+    #[test]
+    fn delete_records_returns_error_when_input_records_are_invalid() {
+        let home_dir = ensure_node_home_directory_exists(
+            "sent_payable_dao",
+            "delete_records_returns_error_when_input_records_are_invalid",
+        );
+        let wrapped_conn = DbInitializerReal::default()
+            .initialize(&home_dir, DbInitializationConfig::test_default())
+            .unwrap();
+        let subject = SentPayableDaoReal::new(wrapped_conn);
+
+        let result = subject.delete_records(&HashSet::new());
+
+        assert_eq!(result, Err(SentPayableDaoError::EmptyInput));
+    }
+
+    #[test]
+    fn delete_records_returns_error_when_no_records_are_deleted() {
+        let home_dir = ensure_node_home_directory_exists(
+            "sent_payable_dao",
+            "delete_records_returns_error_when_no_records_are_deleted",
+        );
+        let wrapped_conn = DbInitializerReal::default()
+            .initialize(&home_dir, DbInitializationConfig::test_default())
+            .unwrap();
+        let subject = SentPayableDaoReal::new(wrapped_conn);
+        let non_existent_hash = H256::from_low_u64_le(999);
+        let hashset = HashSet::from([non_existent_hash]);
+
+        let result = subject.delete_records(&hashset);
+
+        assert_eq!(result, Err(SentPayableDaoError::NoChange));
+    }
+
+    #[test]
+    fn delete_records_returns_error_when_not_all_input_records_were_deleted() {
+        let home_dir = ensure_node_home_directory_exists(
+            "sent_payable_dao",
+            "delete_records_returns_error_when_not_all_input_records_were_deleted",
+        );
+        let wrapped_conn = DbInitializerReal::default()
+            .initialize(&home_dir, DbInitializationConfig::test_default())
+            .unwrap();
+        let subject = SentPayableDaoReal::new(wrapped_conn);
+        let present_hash = H256::from_low_u64_le(1);
+        let absent_hash = H256::from_low_u64_le(2);
+        let tx = TxBuilder::default()
+            .hash(present_hash)
+            .status(TxStatus::Failed)
+            .build();
+        subject.insert_new_records(&vec![tx]).unwrap();
+        let hashset = HashSet::from([present_hash, absent_hash]);
+
+        let result = subject.delete_records(&hashset);
+
+        assert_eq!(
+            result,
+            Err(SentPayableDaoError::PartialExecution(
+                "Only 1 of the 2 hashes has been deleted.".to_string()
+            ))
+        );
+    }
+
+    #[test]
+    fn delete_records_returns_deletion_failed_error_when_an_error_occurs_in_sql() {
```

Comment:
maybe "delete_records_returns_a_general_error_from_sql"

Please make it match with the name used in ensure_node_home_directory_exists in this test.